doc/chips/11.rst

```diff
@@ -0,0 +1,183 @@
+we considered a variety of features. Some of them were accepted
+and some of them were rejected. This document records some of
+the discussed-and-rejected features for posterity.
+
+Description
+-----------
+
+Alternative syntax for distinguishing Phase 1 and Phase 2 in an initializer
+***************************************************************************
+
+We considered providing dedicated syntax for distinguishing Phase 1
+and Phase 2 in an initializer. This syntax, codenamed Syntax 2,
+designates different block statements for Phase 1 and Phase 2,
+separated by the proposed "finalize" keyword.
+
+.. code-block:: chapel
+
+   proc init() {
+     ... Phase 1 code ...
+     super.init(...);
+   } finalize {
+     ... Phase 2 code ...
+   }
+
+Certain pieces of this syntax could be omitted as follows:
+
+* The super.init() call, if it takes no arguments.
+
+* The Phase 1 block statement, if contains no statements.
+  In this case, the "finalize" keyword can also be omitted.
+  Default field initializations followed by super.init() call
+  with no arguments occur in this case.
+
+Here is an example of an initializer where only Phase 2 is
+specified explicitly:
+
+.. code-block:: chapel
+
+   proc init() {
+     ... Phase 2 code ...
+   }
+
+Syntax 2 has the advantage of visually-clear separation of the two
+phases. We did not incorporate it because it did not allow sharing
+variables or extending an if statement across the two phases.
+
+Child's fields could depend on parent's fields
+**********************************************
+
+We considered allowing child's fields to be initialized based on
+parent's fields. Below is one way to achieve this.
+
+The child's initializer could control where, in its Phase 1, the
+parent's Phase 1 is invoked. In other words, the parent's Phase 1
+would not necessarily come at the end of the child initializer's Phase 1.
+The following example shows that based on Syntax 2.
+
+.. code-block:: chapel
+
+   class Parent {
+     var D: domain(1);
+     proc init(myRange) {
+       D = {myRange};  // Phase 1
+     } finalize {
+       ... Phase 2 code ...
+     }
+   }
+
+   class Child: Parent {
+     var xD: domain(1);
+     proc init(myRange, extra) {
+       super.init(myRange);   // initializes Parent.D
+       xD = D.expand(extra);  // initialize xD based on Parent.D
+     } finalize {
+       ... Phase 2 code ...
+     }
+   }
+
+Analogously, we could allow the child initializer to control the point
+during its Phase 2 when the parent initializer's Phase 2 is to be invoked.
+For example:
+
+.. code-block:: chapel
+
+   class Child: Parent {
+     proc init(...) {
+       ... Phase 1 code ...
+     } finalize {
+       ... Phase 2 in Child ...
+       super.finalize();      // executes Phase 2 in Parent
+     }
+   }
+
+We did not incorporate this functionality because it adds complexity to
+how initialization occurs.
+
+
+Parent could complete initialization after child's initialization is complete
```

Comment:
I still object to including this bullet.  It never felt seriously considered, and I feel this CHIP should only handle the seriously considered options (or reading it would not be worth someone's time)